__docs__/wonder-blocks-switch/switch-best-practices.stories.tsx

```diff
@@ -31,6 +31,13 @@ export default {
             disableSnapshot: true,
         },
     },
+    decorators: [
+        (Story) => (
+            <RenderStateRoot>
+                <Story />
+            </RenderStateRoot>
+        ),
+    ],
```

Comment:
**question:** What do you think about adding this globally to all stories in `./storybook/preview.tsx`?

Your solution works fine for the Switch case, if you prefer to keep this PR only scoped for switch.

Reply:
Make sure every story works in that case. By default, nested `RenderStateRoot` instances will throw an error so any story that is explicitly rendering one is going to break if you wrap them all by default. A story that is specifically using `RenderStateRoot` as an example but doesn't include it in the story would be weird, as would one that sets the `throwIfNested` value to `false` but we wanted to give a default, recommended usage story.

Reply:
i took some inspiration from here https://github.com/storybookjs/storybook/issues/23237#issuecomment-1611351405 and added a global param that defaults to true. let me know you guys are ok w this solution or would prefer i just refactor the stories.

Reply:
Yeah, that's a good compromise, @marianmwang - I like that. :)